jetstream/external_config.py

```diff
@@ -39,6 +46,99 @@ def validate(self, experiment: jetstream.experimenter.Experiment) -> None:
         Analysis("no project", "no dataset", conf).validate()
 
 
+def check_all_keys_lowercase(config: MutableMapping[str, Any]) -> Dict[Any, Any]:
+    """
+    function to recursively check that configuration keys are lowercase.
+
+    Returns a dictionary with a dict key path to key which is invalid.
+
+    """
+
+    invalid_keys = dict()
+    for key, value in config.items():
+        if isinstance(value, dict):
+            result = check_all_keys_lowercase(value)
+            if result:
+                invalid_keys[key] = result
+
+        if key.lower() != key:
+            invalid_keys[key] = {key: "INVALID_CONFIG_SETTING"}
+
+    return invalid_keys
+
+
+def validate_config_settings(config_file: Path) -> None:
+    """
+    Loads external config file and runs a number of validation steps on it:
+    - Checks that all config keys/settings are lowercase
+    - Checks for missing core config keys
+    - Checks for unexpected core configuration keys
+    - Checks that all segments defined under experiment have configuration in segments section
+    - Checks if metric with custom config is defined in metrics.weekly or metrics.overall fields
+
+    Returns None, if issues found with the configuration an Exception is raised
+    """
+
+    config = toml.loads(config_file.read_text())
+
+    if config_errors := check_all_keys_lowercase(config):
+        err_msg = (
+            "It appears some config settings/keys"
+            f"contain unexpected uppercase letters: {config_errors}"
+        )
+        raise WrongCaseConfigurationException(err_msg)
+
+    required_core_config_keys = (
+        "experiment",
+        "segments",
+        "metrics",
```

Comment:
```suggestion
        "metrics",
```

The only key that is required is `metrics`.

`experiment` is only required when some Experiment configurations need to be overwritten, otherwise we simply use what is configured in Experimenter.

`segments` are optional. If none are specified then all users will be part of the same segment